lib/animina_web/components/bookmark/bookmark_component.ex

```diff
@@ -55,6 +55,17 @@ defmodule AniminaWeb.BookmarkComponent do
     end
   end
 
+  # we display the state as an atom in the photo struct, but we need to make sure it is an atom
+  # as we are using it for pattern matching
+
+  defp make_sure_photo_state_is_atom(nil) do
+    ""
+  end
+
+  defp make_sure_photo_state_is_atom(photo) do
```

Comment:
What happens if this already is an atom? Or are we sure that this can not happen?

Reply:
We have this is_atom(@sender.profile_photo.state)

Reply:
> We have this is_atom(@sender.profile_photo.state)

So we check it in two different places? Wouldn't it be better to do this in one place/area?

Reply:
It would , I will refactor this code , already an issue for it